app/lib/job_feed_importers/talroo_job_feed_importer.rb

```diff
@@ -0,0 +1,143 @@
+module JobFeedImporters
+  class TalrooJobFeedImporter
+    URL     = ENV['TALROO_URL']
+    BUFFER_LIMIT = 10 * 1024 * 1024
+    OPTIONS = {
+      headers: {
+        'Content-Type' => 'text/xml',
+        'Accept'       => 'text/xml',
+      },
+      accept_encoding: 'gzip',
+    }.freeze
+
+    def self.start(*tags)
+      new(*tags).import
+    end
+
+    def initialize(*tags)
+      @tags = tags.flatten
+    end
+
+    def import
+      purge_jobs
+
+      job_buffer = ''
+      req = Typhoeus::Request.new(URL, OPTIONS)
+      req.on_body do |chunk|
+        job_buffer << chunk
+        if job_buffer.size > BUFFER_LIMIT
+          jobs = process_buffer(job_buffer)
+          JobPosting.bulk_insert values: jobs if jobs.present?
+          job_buffer = ''
+        end
+      end
+      req.run
+    end
+
+    private
+
+    def purge_jobs
+      JobPosting.talroo.delete_all
```

Comment:
we have links to jobs.. if we completely delete them all, we'll need to make sure we use the remote unique ID in the URL or something like that.

For example, if we import a job and the URL is '/jobs/123', then we reimport the same job the next day, we want the URL to be the same for the same job.

If this adds scope, please let me know.

Reply:
Easy to accomplish this by defining `to_param` on the `JobPosting` model that has smarts about remotely sourced jobs.